src/contrib/mlir/tools/ngraph-opt/ngraph_opt.cpp

```diff
@@ -0,0 +1,81 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/check.hpp"
+
+#include <llvm/Support/CommandLine.h>
+#include <llvm/Support/MemoryBuffer.h>
+#include <llvm/Support/ToolOutputFile.h>
+#include <mlir/Pass/Pass.h>
+#include <mlir/Pass/PassManager.h>
+#include <mlir/Support/FileUtilities.h>
+#include <mlir/Support/MlirOptMain.h>
+
+static llvm::cl::opt<std::string>
+    input_filename(llvm::cl::Positional, llvm::cl::desc("<input file>"), llvm::cl::init("-"));
+
+static llvm::cl::opt<std::string> output_filename("o",
+                                                  llvm::cl::desc("Output filename"),
+                                                  llvm::cl::value_desc("filename"),
+                                                  llvm::cl::init("-"));
+
+static llvm::cl::opt<bool>
+    split_input_file("split-input-file",
+                     llvm::cl::desc("Split the input file into pieces and process each "
+                                    "chunk independently"),
+                     llvm::cl::init(false));
+
+static llvm::cl::opt<bool>
+    verify_diagnostics("verify-diagnostics",
+                       llvm::cl::desc("Check that emitted diagnostics match "
+                                      "expected-* lines on the corresponding line"),
+                       llvm::cl::init(false));
+
+static llvm::cl::opt<bool>
+    verify_passes("verify-each",
+                  llvm::cl::desc("Run the verifier after each transformation pass"),
+                  llvm::cl::init(true));
+
+static std::vector<const mlir::PassRegistryEntry*>* pass_list;
+
+int main(int argc, char** argv)
```

Comment:
Could you add some documentation for the tool and its CLI?

Reply:
Sorry, I was focused on something else and the patch went in :). I'll address this in a separate commit.